### PR TITLE
[feature](Cloud) Support drop storage vault in Meta Service

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 1]
```diff
@@ -322,16 +328,32 @@ static int add_hdfs_storage_valut(InstanceInfoPB& instance, Transaction* txn,
...
+    LOG_INFO("try to put storage vault_id={}, vault_name={}", vault_id, hdfs_param.name());
     instance.mutable_resource_ids()->Add(std::move(vault_id));
     *instance.mutable_storage_vault_names()->Add() = hdfs_param.name();
     return 0;
 }
 
-// TODO(ByteYue): Implement drop storage vault.
-[[maybe_unused]] static int remove_hdfs_storage_valut(std::string_view vault_key, Transaction* txn,
-                                                      MetaServiceCode& code, std::string& msg) {
+static int remove_hdfs_storage_valut(InstanceInfoPB& instance, Transaction* txn,
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:337]: typo remove_hdfs_storage_valut

File: cloud/test/meta_service_test.cpp [hunk 2]
```diff
@@ -5088,6 +5082,113 @@ TEST(MetaServiceTest, AddHdfsInfoTest) {
...
+
+        brpc::Controller cntl;
+        AlterObjStoreInfoResponse res;
+        meta_service->alter_obj_store_info(
+                reinterpret_cast<::google::protobuf::RpcController*>(&cntl), &req, &res, nullptr);
+        ASSERT_EQ(res.status().code(), MetaServiceCode::OK) << res.status().msg();
+        InstanceInfoPB instance;
+        get_test_instance(instance);
+        ASSERT_EQ(instance.resource_ids().size(), 0);
+        ASSERT_EQ(instance.storage_vault_names().size(), 0);
```
Comment (gavinchou) [inline cloud/test/meta_service_test.cpp:5185]: read the InstanceInfoPB again and read  the removed key to ensure all things go right

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Review by gavinchou — APPROVED** [review]

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**